webapp/src/hooks/usePluginApi.ts

```diff
@@ -1,28 +1,39 @@
 import {useSelector, useDispatch} from 'react-redux';
+
 import {AnyAction} from 'redux';
 
+import {setApiRequestCompletionState} from 'reducers/apiRequest';
+
 import services from 'services';
 
 function usePluginApi() {
     const state = useSelector((pluginState: PluginState) => pluginState);
     const dispatch = useDispatch();
 
     // Pass payload only in POST rquests for GET requests there is no need to pass payload argument
-    const makeApiRequest = (serviceName: ApiServiceName, payload: APIRequestPayload): Promise<AnyAction> | any => {
+    const makeApiRequest = async (serviceName: ApiServiceName, payload: APIRequestPayload): Promise<AnyAction> | any => {
         return dispatch(services.endpoints[serviceName].initiate(payload)); //TODO: add proper type here
```

Comment:
Is there still need for this TODO comment?

Reply:
yes, this will be removed at the end when we would be not using this at all